src/middlewared/middlewared/plugins/pool_/unlock.py

```diff
@@ -87,9 +87,11 @@ async def restart_vms_after_unlock(self, dataset_name):
     @private
     async def restart_services_after_unlock(self, dataset_name, services_to_restart):
         try:
-            await self.middleware.call('core.bulk', 'service.restart', [
-                [i] for i in set(services_to_restart) - {'jails', 'vms'}
-            ])
+            to_restart = [[i] for i in set(services_to_restart) - {'jails', 'vms'}]
+            if not to_restart:
+                return
+
+            await self.middleware.call('core.bulk', 'service.restart', to_restart)
```

Comment:
@sonicaj Should we wait core.bulk to finish?

Reply:
I think we should, this would ensure that once the unlock job completes we have necessary services restarted/reloaded.